scripts/run/update.sh

```diff
@@ -1,78 +1,241 @@
+
+  return 0
 }
 
-function parse_arguments() {
-  while getopts ":b:" opt; do
-    case ${opt} in
-      b)
-        GIT_BRANCH="${OPTARG}"
-        ;;
-      \?)
-        echo "Invalid option: -${OPTARG}" >&2
-        print_usage
-        ;;
-      :)
-        echo "Option -${OPTARG} requires an argument." >&2
-        print_usage
-        ;;
-    esac
+#######################################
+# Checks the status of all services defined in the `templates` directory.
+# If a service is not active, attempts to start it.
+# Globals:
+#   CWD - Directory containing service templates.
+# Arguments:
+#   None
+# Outputs:
+#   Status messages regarding the activity of each service and attempts to
+#   start inactive services.
+#######################################
+function check_system_status() {
+  echo "Checking system status..."
+
+  for template in "${CWD}"/../../templates/*.service; do
+    declare -r servname=$(basename "${template}")
+    
+    status=$(systemctl is-active "${servname}")
+    if [ "${status}" != "active" ]; then
+      echo "${WARNING} ${YELLOW}'${servname}' is not active.${RESET}"
+      
+      echo "Attempting to start..."
+      systemctl start "${servname}"
+
+      if ! systemctl is-active --quiet "${servname}"; then
+        echo "${FAIL} ${RED}Unable to start '${servname}'${RESET}"
+        rollback_changes || echo "${CRITICAL} ${RED}Unable to rollback changes!${RESET}" 
+        return 1
+      fi
+    fi
+    echo "${OK} '${servname}' is active."
   done
+
+  return 0
+}
+
+#######################################
+# Rolls back the most recent changes by resetting the HEAD to the previous
+# commit and then restores the contents of the service files.
+# Globals:
+#   None
+# Arguments:
+#   None
+# Outputs:
+#   Warning message about rolling back changes and the results of updating
+#   services after the rollback.
+# Returns:
+#   Status of update_services function.
+#######################################
+function rollback_changes() {
+  echo "${WARNING} ${YELLOW}Rolling back changes...${RESET}"
+
+  git reset --hard HEAD~1
+  return "$(update_services)"
 }
 
+#######################################
+# Main function to ensure network connectivity, pull latest changes from a
+# specified branch, update service files, and check the system status. If
+# the network is down, it attempts to connect to WiFi.
+# Globals:
+#   OK, GREEN, WARNING, YELLOW, FAIL, RED, RESET - Used for coloring messages.
+# Arguments:
+#   None
+# Outputs:
+#   Various status messages about network connectivity, Git operations, service
+#   file updates, and system status checks.
+#######################################
 function main() {
-  local scripts=("post-reboot/set_user_permissions.sh" "pre-reboot/create_services.sh")
+  if wget -q --spider "https://google.com"; then
+    echo "${OK} ${GREEN}Network connection successful.${RESET}"
+  else
+    echo "${WARNING} ${YELLOW}Network connection failed.${RESET}"
+
+    declare response
```

Comment:
Where is this response used? What is the behavior when it is `Nn`? Does the return statement cause the main function to stop prematurely -- Do the updates still happen?

Reply:
The response is used to store the response from the user when asked if they want to connect to the network in case a network connection has not been established before running the script. You can see its use in lines 191 and 192. Using the declare keyword is mostly for readability, as I could have simply initialized response in the read statement.

Reply:
When the user indicates that they do not want to connect to a network (for some reason) by typing `N` or `n`, the update scripts exits as a network connection is required. The only case in which a execution path gets to this point is if the device is not connected to the network.